plugin/metrics/prometheus/options.go

```diff
@@ -64,7 +67,8 @@ func (opt *Options) AddFlags(flagSet *flag.FlagSet) {
 	nsConfig := &opt.Primary
 	flagSet.String(nsConfig.namespace+suffixServerURL, defaultServerURL, "The Prometheus server's URL, must include the protocol scheme e.g. http://localhost:9090")
 	flagSet.Duration(nsConfig.namespace+suffixConnectTimeout, defaultConnectTimeout, "The period to wait for a connection to Prometheus when executing queries.")
-
+	flagSet.Bool(nsConfig.namespace+suffixAllowedTokenFromContext, defaultAllowedTokenFromContext, "Flag to read token from context. When set, bearer token from context is used which also overrides token from file")
```

Comment:
@albertteoh 

Can i use another key for configuration like this 

`const StoragePropagationKey = "storage.propagate.token"`

`const PrometheusPropagationKey = "prometheus.propagate.token"`


and set key here

`
			// TODO: Need to figure out set enable/disable propagation on storage and prometheus plugins.
			v.Set(bearertoken.StoragePropagationKey, queryOpts.BearerTokenPropagation)
			v.Set(bearertoken.PrometheusPropagationKey, queryOpts.BearerTokenPropagation)
`

then finally in InitFromViper()

set 	

`cfg.AllowTokenFromContext = v.GetBool(bearertoken.PrometheusPropagationKey)`

Does this look like a step in right direction ?

Reply:
You raise a good point. 

Firstly, my understanding of the motivation behind why bearer token propagation was introduced is for access to the raw trace data through directly hitting the Jaeger Query API that is backed by ES storage requiring authentication: https://github.com/jaegertracing/jaeger/pull/1599. That is, Jaeger UI is no longer the sole client of Jaeger Query for the storage use case (specifically ES at least).

So far, the _only_ use case that I'm aware of for Jaeger Query's suite of `/metrics/...` endpoints is purely for the SPM feature, meaning that Jaeger UI is the _only_ client for these endpoints. This, in turn, means that there shouldn't be a need for bearer token propagation within the `metrics/prometheus` plugin (or more generally, any `metrics` plugin). If that's the case, we could omit any concerns relating to token propagation and just focus on tokens supplied from file for Prometheus.

What I concluded above is just from reading code and past PRs; @yurishkuro feel free to jump in if you have some additional insights based on past experience that run counter to the conclusion above.

Reply:
@albertteoh if i understand correctly, since there isn't need to bearer token propagation like it was required in #1599. 
`cfg.AllowTokenFromContext` in plugin/metrics/prometheus/options.go can be set to `False` and we can just focus on reading token from file path.

Reply:
> `cfg.AllowTokenFromContext` in plugin/metrics/prometheus/options.go can be set to False

I think there should not be `cfg.AllowTokenFromContext` in `plugin/metrics/prometheus/options.go` at all.

> we can just focus on reading token from file path.

Yes, I think we should just focus on reading token from file path, which is sufficient for the requirements detailed in: https://github.com/jaegertracing/jaeger/issues/4219. We can always add token propagation as a separate PR if there is a future need for it. For now, I think the YAGNI rule prevails.